### ISSUE TITLE
Development: can't call wesnoth.wml_actions.message twice

### ISSUE THREAD

**Issue by vgaming** [wesnoth/wesnoth#2777]
Steps:
* start a new game
* open Lua console
* type `wesnoth.wml_actions.message{message="test"}`
* press Esc
* type the previous command again (`wesnoth.wml_actions.message{message="test"}`)

Expected at this point: generate a new message.

Actually: no message is generated, wesnoth is skipping messages: https://wiki.wesnoth.org/LuaWML:Display#wesnoth.skip_messages

This is not a bug because nobody promised a particular behavior from Lua console.
Still, for a developer, it would be quite useful to have Lua-console-dependent events considered "separate" with skipping messages flag being reset.
Thoughts?

**Comment by CelticMinstrel** [comment]
I'd say if you're calling it from the console... don't press Esc? And IIRC there's a Lua function/attribute for the skipping flag in any case, so you could reset it manually and then get another message.

**Comment by CelticMinstrel** [comment]
```lua
function debug_message(cfg)
    wesnoth.wml_actions.message(cfg)
    wesnoth.skip_messages(false)
end
```

**Comment by vgaming** [comment]
@CelticMinstrel yes, you can work-around this issue by re-setting `wesnoth.skip_messages(false)` manually (I linked it myself in the issue). The question is: are there any reasons not to do that by default?

**Comment by CelticMinstrel** [comment]
What, you mean having the Lua console call `wesnoth.skip_messages(false)` after every command? I suppose you could do that....

**Comment by vgaming** [comment]
@CelticMinstrel yes, or make the actions belong to different "events". As @ln-zookeeper told in IRC chat, the flag is normally reset when an event ends and a new begins.
Makes total sense to have separate events for separate commands in Lua console.

**Comment by CelticMinstrel** [comment]
Commands from the Lua console don't belong to any event at all (unless the Lua console was invoked during an event via `wesnoth.show_lua_console`).

**Comment by vgaming** [comment]
I guess this is controversial, closing.

**Comment by jostephd** [comment]
I [don't hear](https://github.com/wesnoth/wesnoth/issues/2777#issuecomment-377539467) a controversy, and I agree that calling message, Esc, message should print two messages.

**Comment by vgaming** [comment]
@jostephd Well it'd be natural as well to only execute the Lua that you explicitly written down, and nothing besides that. What if, for example, you want to test out this `Esc` thing? Then the behavior of the Lua console would break your experimentation.

From this point of view, the decision seems subjective to me, and you could go both ways with that. I'm OK with keeping the issue open & changing the behavior as well though. It certainly shouldn't break add-ons.

**Comment by jostephd** [comment]
> What if, for example, you want to test out this `Esc` thing itself? Then the behavior of the Lua console would break your experimentation.

Then you could call `wesnoth.wml_actions.message{message="test"}; wesnoth.wml_actions.message{message="test"}` in one line. (Right now that redraws the console under the second message, which is probably a bug?)

At the very least, the 'skip messages' bit should be reset when the lua console is closed and reopened...
